src/time_steppers.jl

```diff
@@ -220,12 +176,34 @@ function update_source_terms!(::Val{Dev}, fCor, χ, ρ₀, κh, κv, 𝜈h, 𝜈
-                @inbounds Gw[i, j, k] = -u∇w(u, v, w, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)                                                                           + 𝜈∇²w(w, 𝜈h, 𝜈v, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k) + F.w(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
-
-                @inbounds GT[i, j, k] = -div_flux(u, v, w, T, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k) + κ∇²(T, κh, κv, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k) + F.T(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
-                @inbounds GS[i, j, k] = -div_flux(u, v, w, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k) + κ∇²(S, κh, κv, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k) + F.S(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                # u-momentum equation
+                @inbounds Gu[i, j, k] = (-u∇u(u, v, w, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + fCor*avg_xy(v, Nx, Ny, i, j, k)
+                                            - δx_c2f(pHY′, Nx, i, j, k) / (Δx * ρ₀)
+                                            + 𝜈∇²u(u, 𝜈h, 𝜈v, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + F.u(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k))
+
+                # v-momentum equation
+                @inbounds Gv[i, j, k] = (-u∇v(u, v, w, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            - fCor*avg_xy(u, Nx, Ny, i, j, k)
+                                            - δy_c2f(pHY′, Ny, i, j, k) / (Δy * ρ₀)
+                                            + 𝜈∇²v(v, 𝜈h, 𝜈v, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + F.v(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k))
+
+                # w-momentum equation: comment about how pressure and buoyancy are handled
+                @inbounds Gw[i, j, k] = (-u∇w(u, v, w, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + 𝜈∇²w(w, 𝜈h, 𝜈v, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + F.w(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k))
+
+                # temperature equation
+                @inbounds GT[i, j, k] = (-div_flux(u, v, w, T, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + κ∇²(T, κh, κv, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k) 
+                                            + F.T(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k))
+
+                # salinity equation
+                @inbounds GS[i, j, k] = (-div_flux(u, v, w, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + κ∇²(S, κh, κv, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k)
+                                            + F.S(u, v, w, T, S, Nx, Ny, Nz, Δx, Δy, Δz, i, j, k))
 
                 @inbounds Gu[i, j, k] = (1.5 + χ)*Gu[i, j, k] - (0.5 + χ)*Gpu[i, j, k]

```

Comment:
Hmmm, I wonder if we can inline the previous lines calculating `Gu`, `Gv`, etc. into here so we'd only have to store the source term from one time step. That would free up so much memory!
```julia
Gu[i, j, k] = (1.5 + χ)*(inlined stuff) - (0.5 + χ)*Gu[i,j,k]
```

Reply:
Interesting idea! It does lock us further lock the algorithm into Adams-Bashforth. However, maybe it makes sense for this software to develop ultra-optimized code for a small number of time-stepping schemes, rather than flexible code for a large number of time-stepping schemes... ?

Is `Gp` used anywhere else?

Reply:
Just checked and it's not used anywhere else so pretty sure we can do this.

Yeah it does lock us in but I don't see us using another time-stepping scheme in the very near future. We know AB2 works rather well (and already uses little memory). And on GPUs we'll be able to store larger problems.